### PR TITLE
Noise listener

### PR DESCRIPTION
**Pull request by cheezus1**

### CODE REVIEW

File: test/core_listener_test.exs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    receive do
+      {:micro_blocks, _} -> :ok
+    end
+
+    Listener.subscribe(:transactions, self())
+
+    Account.spend(setup_data.client, setup_data.client.keypair.public, 100)
+
+    receive do
+      {:transactions, _} -> :ok
```
Comment (thepiwo) [inline test/core_listener_test.exs]: more checks can be done if this is the expected transaction

File: test/core_listener_test.exs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    end
+
+    Listener.subscribe(:pool_transactions, self())
+
+    Account.spend(setup_data.client, setup_data.client.keypair.public, 100)
+
+    receive do
+      {:pool_transactions, _} -> :ok
```
Comment (thepiwo) [inline test/core_listener_test.exs]: more checks can be done if this is the expected transaction

File: test/core_listener_test.exs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    Listener.start(
+      ["aenode://pp_rr2Bz9zFDA78u3daZxZVyGvHazafc43PxQaXTcc5nwRj41sfc@localhost:3015"],
+      "my_test",
+      "kh_2KhFJSdz1BwrvEWe9fFBRBpWoweoaZuTiYLWwUPh21ptuDE8UQ"
+    )
+
+    Listener.subscribe(:key_blocks, self())
+
+    receive do
+      {:key_blocks, _} -> :ok
```
Comment (thepiwo) [inline test/core_listener_test.exs]: does this logic fail the tests if nothing is received? How is the listener stopped in the tests?

Comment (d-velev) [reply test/core_listener_test.exs]: Added a 2 second timeout to receives and the listener is stopped at the end

File: lib/utils/governance.ex [hunk 4]
```diff
@@ -1,7 +1,5 @@
 defmodule Utils.Governance do
-  @moduledoc """
-  Contains all constants and helper functions, related to blockchain.
-  """
+  @moduledoc false
```
Comment (DanielaIvanova) [inline lib/utils/governance.ex]: Why the documentation is removed?

File: lib/utils/serialization_utils.ex [hunk 5]
```diff
@@ -1,7 +1,5 @@
 defmodule Utils.SerializationUtils do
-  @moduledoc """
-  Serialization helper module.
-  """
+  @moduledoc false
```
Comment (DanielaIvanova) [inline lib/utils/serialization_utils.ex]: Same as above.

File: lib/utils/serialization_utils.ex [hunk 6]
```diff
@@ -33,36 +31,6 @@ defmodule Utils.SerializationUtils do
 
   @ct_version 0x40001
```
Comment (DanielaIvanova) [inline lib/utils/serialization_utils.ex]: Same as above.

File: lib/core/listener/peer_connection.ex [hunk 7]
```diff
@@ -0,0 +1,485 @@
+defmodule Core.Listener.PeerConnection do
+  @moduledoc false
```
Comment (DanielaIvanova) [inline lib/core/listener/peer_connection.ex]: Why the documentation is set to `false`?

File: lib/utils/governance.ex (hunk 4 above)
Comment (d-velev) [reply lib/utils/governance.ex]: This module wouldn't make sense to users and is used internally

**Comment by d-velev** [comment]
@DanielaIvanova I've set `@moduledoc false` on modules that are used internally and don't make sense to users

File: lib/core/listener/listener.ex [hunk 8]
```diff
@@ -0,0 +1,464 @@
...
+      {:error, _} = err ->
+        err
+    end
+  end
+
+  def unsubscribe(event, _, _), do: {:error, "Unknown event to unsubscribe from: #{event}"}
+
+  @doc false
+  def notify(event, data, hash) when event in @notifiable_events do
+    # IO.inspect({event, data, hash}, limit: :infinity)
```
Comment (DanielaIvanova) [inline lib/core/listener/listener.ex]: This should be removed.

File: lib/utils/serialization.ex [hunk 9]
```diff
@@ -250,6 +254,428 @@ defmodule Utils.Serialization do
...
+       ) do
+    %{
+      channel_id: :aeser_api_encoder.encode(:id_hash, channel_id),
+      from_id: :aeser_api_encoder.encode(:id_hash, from_id),
+      payload: :aeser_api_encoder.encode(:transaction, payload),
+      round: round,
+      # TODO
+      update: update,
+      state_hash: :aeser_api_encoder.encode(:state, state_hash),
+      # TODO
```
Comment (DanielaIvanova) [inline lib/utils/serialization.ex]: Is it still `TODO`?

Comment (cheezus1) [reply lib/utils/serialization.ex]: yes, together with the POI serialization

**Comment by thepiwo** [comment]
@d-velev @cheezus1 merge whenever you want

**Comment by thepiwo** [comment]
feel free to merge

**Pull request merged**